internal/controller/stages/verification.go

```diff
@@ -224,6 +251,54 @@ func (r *reconciler) getVerificationInfo(
 	}
 }
 
+func (r *reconciler) abortVerification(
+	ctx context.Context,
+	stage *kargoapi.Stage,
+) *kargoapi.VerificationInfo {
+	if r.rolloutsClient == nil {
+		return &kargoapi.VerificationInfo{
+			ID:    stage.Status.CurrentFreight.VerificationInfo.ID,
+			Phase: kargoapi.VerificationPhaseError,
+			Message: "Rollouts integration is disabled on this controller; cannot " +
+				"abort verification",
+		}
+	}
+
+	ar := &rollouts.AnalysisRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      stage.Status.CurrentFreight.VerificationInfo.AnalysisRun.Name,
+			Namespace: stage.Status.CurrentFreight.VerificationInfo.AnalysisRun.Namespace,
+		},
+	}
+	if err := r.patchAnalysisRunFn(
+		ctx,
+		ar,
+		client.RawPatch(types.MergePatchType, []byte(`{"spec":{"terminate":true}}`)),
+	); err != nil {
+		return &kargoapi.VerificationInfo{
+			ID:    stage.Status.CurrentFreight.VerificationInfo.ID,
+			Phase: kargoapi.VerificationPhaseError,
+			Message: fmt.Errorf(
+				"error terminating AnalysisRun %q in namespace %q: %w",
+				ar.Name,
+				ar.Namespace,
+				err,
+			).Error(),
+		}
+	}
+
+	// Return a new VerificationInfo with the same ID and a message indicating
+	// that the verification was aborted. The Phase will be set to Failed, as
+	// the verification was not successful.
+	// We do not use the further information from the AnalysisRun, as this
+	// will indicate a "Succeeded" phase due to Argo Rollouts behavior.
+	return &kargoapi.VerificationInfo{
+		ID:      stage.Status.CurrentFreight.VerificationInfo.ID,
+		Phase:   kargoapi.VerificationPhaseFailed,
```

Comment:
I think we should introduce an Aborted phase, then there are three distinct and non-overlapping negative outcomes:

* Kargo encountered a problem (error)
* Verification completed with a negative result (failure)
* Verification was aborted

wdyt?

Reply:
This crossed my mind as well, so more than happy to add it!